New section on attaching custom metadata to Prometheus time series (#23203)
openshift/openshift-docs: modules/monitoring-attaching-additional-labels-to-your-time-series.adoc

// Module included in the following assemblies:
//
// * monitoring/cluster_monitoring/configuring-the-monitoring-stack.adoc

[id="attaching-additional-labels-to-your-time-series_{context}"]

> = Attaching additional labels to your time series

Reviewer: I would add alerts too since it's most likely what users want to customize: "Attaching additional labels to time series and alerts"

Using the external labels feature of Prometheus, you can attach additional custom labels to all time series and alerts leaving the Prometheus cluster.

.Prerequisites

> * Make sure you have the `cluster-monitoring-config` ConfigMap object with the `data/config.yaml` section.

Reviewer: It's not clear to me what to do here. Maybe have a one-liner command to create an empty ConfigMap?

```
oc create -n openshift-monitoring configmap cluster-monitoring-config --from-literal 'config.yml={}'
```

Author: @simonpasquier This is the convention we use in every section that involves editing `cluster-monitoring-config`, beginning with [1] . Ideally, we would link to [2], which answers your question, but we have to resort to just referring to it with this prerequisite line without linking.
[1] https://docs.openshift.com/container-platform/4.4/monitoring/cluster_monitoring/configuring-the-monitoring-stack.html#configuring-the-cluster-monitoring-stack_configuring-monitoring
[2] https://docs.openshift.com/container-platform/4.4/monitoring/cluster_monitoring/configuring-the-monitoring-stack.html#creating-cluster-monitoring-configmap_configuring-monitoring

.Procedure

. Start editing the `cluster-monitoring-config` ConfigMap:
+
----
$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
----

. Define a map of labels you want to add for every metric under `data/config.yaml`:
+
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: cluster-monitoring-config
  namespace: openshift-monitoring
data:
  config.yaml: |
    prometheusK8s:
      externalLabels:

>         <key>: <value>

Reviewer 2: ping @lilic is this still valid for the user workload monitoring config split?

Reviewer 3: This is going to be merged into 4.5 branch, we did not change anything for 4.5.

----
+
Substitute `<key>: <value>` with a map of key-value pairs where `<key>` is a unique name of the new label and `<value>` is its value.
+
[WARNING]
====
Do not use `prometheus` or `prometheus_replica` as key names, because they are reserved and would be overwritten.
====
+

> For example, to add metadata about the region and environment where the time series originate, use:

Reviewer: ```suggestion
For example, to add metadata about the region and environment to all time series and alerts, use:
```
